datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -187,7 +170,7 @@ impl PruningPredicate {
         // in the predicate means we don't know for sure if the row
         // group can be filtered out or not. To maintain correctness
         // the row group must be kept and thus `true` is returned.
-        match self.predicate_expr.evaluate(&statistics_batch)? {
+        match self.predicate_expr.evaluate(&statistics_batch).expect("bug") {
```

Comment:
I think it would be nicer here to return the error rather than panic, but I also think panic'ing is ok

Reply:
ah, this is debug code, I HAVE TO remove this before merging.

Reply:
done